src/Microsoft.AspNetCore.OData/Extensions/ODataRouteBuilderExtensions.cs

```diff
@@ -251,8 +214,7 @@ public static IRouteBuilder Count(this IRouteBuilder builder)
                 throw Error.ArgumentNull("builder");
             }
 
-            DefaultQuerySettings defaultQuerySettings = builder.GetDefaultQuerySettings();
-            defaultQuerySettings.EnableCount = true;
+            builder.ServiceProvider.Count();
```

Comment:
@xuzhg Do we need the variant of Count that accepts a `QueryOptionSetting` parameter? Given this overload [here](https://github.com/OData/WebApi/pull/2035/files#diff-1ac400cb4d772495bb6ba2aa4f28b4b5R159)

Reply:
Thanks. I forgot to change that Count(...) method. Fixed.

---
In reply to: [384452658](https://github.com/OData/WebApi/pull/2035#discussion_r384452658) [](ancestors = 384452658)